pkg/fqdn/name_manager.go

```diff
@@ -213,6 +217,12 @@ func (n *NameManager) UnregisterFQDNSelector(selector api.FQDNSelector) {
 	n.Lock()
 	defer n.Unlock()
 
+	if _, ok := n.allSelectors[selector]; ok {
+		if metrics.FQDNSelectors.IsEnabled() {
+			metrics.FQDNSelectors.Dec()
```

Comment:
Probably a very minor thing, would it better to count these then do `metrics.FQDNSelectors.Set(n)`.  If the prometheus scrape occurs while this is happening it might emit a metric value that never really existed (although, tbf, I can't see why that would be an issue)

Reply:
Great suggestion!